drivers/dma/dma_nxp_sdma.c

```diff
@@ -0,0 +1,470 @@
+	 * Note that SDMA transfer is later manually started by the dma_nxp_sdma_start() */
+	chan_data->transfer_cfg.isEventIgnore = false;
+	chan_data->transfer_cfg.isSoftTriggerIgnore = false;
+	SDMA_SubmitTransfer(&chan_data->handle, &chan_data->transfer_cfg);
+
+	return 0;
+}
+
+static int dma_nxp_sdma_start(const struct device *dev, uint32_t channel)
+{
+	const struct sdma_dev_cfg *dev_cfg = dev->config;
+	struct sdma_dev_data *dev_data = dev->data;
+	struct sdma_channel_data *chan_data;
+
+	if (channel >= FSL_FEATURE_SDMA_MODULE_CHANNEL) {
+		LOG_ERR("dma_nxp_sdma_start() invalid channel %d", channel);
+		return -EINVAL;
+	}
+
+	chan_data = &dev_data->chan[channel];
+
+	SDMA_SetChannelPriority(dev_cfg->base, channel, DMA_NXP_SDMA_CHAN_DEFAULT_PRIO);
+	SDMA_StartChannelSoftware(dev_cfg->base, channel);
+
+	return 0;
+}
+
+static int dma_nxp_sdma_stop(const struct device *dev, uint32_t channel)
+{
+	struct sdma_dev_data *dev_data = dev->data;
+	struct sdma_channel_data *chan_data;
+
+	if (channel >= FSL_FEATURE_SDMA_MODULE_CHANNEL) {
+		LOG_ERR("dma_nxp_sdma_stop() invalid channel %d", channel);
+		return -EINVAL;
+	}
+
+	chan_data = &dev_data->chan[channel];
+
+	SDMA_StopTransfer(&chan_data->handle);
+	return 0;
+}
+
+static int dma_nxp_sdma_get_status(const struct device *dev, uint32_t channel,
+				   struct dma_status *stat)
+{
+	struct sdma_dev_data *dev_data = dev->data;
+	struct sdma_channel_data *chan_data;
+
+	chan_data = &dev_data->chan[channel];
+
+	stat->free = chan_data->stat.free;
+	stat->pending_length = chan_data->stat.pending_length;
+
+	return 0;
+}
+
+static int dma_nxp_sdma_reload(const struct device *dev, uint32_t channel, uint32_t src,
+			       uint32_t dst, size_t size)
+{
+	struct sdma_dev_data *dev_data = dev->data;
+	struct sdma_channel_data *chan_data;
+
+	chan_data = &dev_data->chan[channel];
+
+	if (!size)
+		return 0;
+
+	if (chan_data->direction == MEMORY_TO_PERIPHERAL)
+		dma_nxp_sdma_produce(chan_data, size);
+	else
+		dma_nxp_sdma_consume(chan_data, size);
+
+	return 0;
+}
+
+static int dma_nxp_sdma_get_attribute(const struct device *dev, uint32_t type, uint32_t *val)
+{
+	switch(type) {
+		case DMA_ATTR_BUFFER_SIZE_ALIGNMENT:
+			*val = 4;
+			break;
+		case DMA_ATTR_BUFFER_ADDRESS_ALIGNMENT:
+			*val = 128; /* should be dcache_align */
+			break;
+		case DMA_ATTR_MAX_BLOCK_COUNT:
+			*val = DMA_NXP_SDMA_BD_COUNT;
+			break;
+		default:
+			LOG_ERR("invalid attribute type: %d", type);
+			return -EINVAL;
+	}
+	return 0;
+}
+
+static bool sdma_channel_filter(const struct device *dev, int chan_id, void *param)
+{
+	struct sdma_dev_data *dev_data = dev->data;
+
+	/* chan 0 is reserved for boot channel */
+	if (chan_id == 0)
+		return false;
+
+	if (chan_id >= FSL_FEATURE_SDMA_MODULE_CHANNEL)
+		return false;
+
+	dev_data->chan[chan_id].event_source = *((int*)param);
+	dev_data->chan[chan_id].index = chan_id;
+
+	return true;
+}
+
+static const struct dma_driver_api sdma_api = {
```

Comment:
All device drivers need to use the `DEVICE_API` macro.

```suggestion
static DEVICE_API(dma, sdma_api) = {
```

Reply:
thanks will fix.